usecases/replica/coordinator.go

```diff
@@ -164,7 +165,9 @@ func (c *coordinator[T]) Push(ctx context.Context,
 		return nil, 0, fmt.Errorf("%w : class %q shard %q", err, c.Class, c.Shard)
 	}
 	level := state.Level
-	nodeCh := c.broadcast(ctx, state.Hosts, ask, level)
+	//nolint:govet // we expressely don't want to cancel that context as the timeout will take care of it
```

Comment:
Isn't it safe to cancel the ctxWithTimeout after broadcast() is done. I believe also that we need to clear the context anyway to not leak resources

Reply:
Discussed live: Moving the context to the callee side is not possible because the callee would return once we have reached the consistency level for the request. So we will keep it not cancelled but with the timeout to ensure *if* we leak a goroutine, it doesn't leak forever.